packages/next/build/utils.ts

```diff
@@ -648,13 +651,18 @@ export async function buildStaticPaths(
             } in getStaticPaths for ${page}`
           )
         }
-
-        builtPage = builtPage.replace(
-          `[${repeat ? '...' : ''}${validParamKey}]`,
-          repeat
-            ? (paramValue as string[]).map(encodeURIComponent).join('/')
-            : encodeURIComponent(paramValue as string)
-        )
+        let replaced = `[${repeat ? '...' : ''}${validParamKey}]`
+        if (optional) {
+          replaced = `[${replaced}]`
+        }
+        builtPage = builtPage
+          .replace(
+            replaced,
+            repeat
+              ? (paramValue as string[]).map(encodeURIComponent).join('/')
+              : encodeURIComponent(paramValue as string)
+          )
+          .replace(/(?!^)\/$/, '')
```

Comment:
What's the purpose of this regex?

Reply:
It removes that trailing slash in case `paramValue` is empty. There may be a cleaner way.

```js
        builtPage
          .replace(
            '/get-static-paths/[[...slug]]',
            [].map(encodeURIComponent).join('/')
          )
        //  '/get-static-paths/'
```

Reply:
Could we also filter the above array?
```js
(paramValue as string[]).filter(Boolean).map(encodeURIComponent).join('/')
```

Reply:
the array is already empty, it's the slash between `/get-static-paths` and `[[...slug]]`.